### PR TITLE
Downgraded minSdk version

### PR DESCRIPTION
**Pull request by KitsuneFolk**
1.  Changed the library's and the example's minSdk to 21(from 23 and 31 respectively). I also created `Theme.kt` which contains `LightTheme` and `DarkTheme` for android version <12 because previously there were used dynamic themes that threw an Exception
2. Updated `androidx-compose-ui-material3`

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 [versions]
-minSdk = "23"
+minSdk = "21"
 compileSdk = "34"
 kotlin = "1.9.10"
 agp = "8.1.0"
 androidx-compose-ui = "1.5.1"   # https://developer.android.com/jetpack/androidx/releases/compose-ui
-androidx-compose-ui-material3 = "1.1.1"
+androidx-compose-ui-material3 = "1.1.2"
```
Comment (saket) [inline gradle/libs.versions.toml]: Can you undo this change? It's best to keep unrelated changes separate from a PR.

File: sample/build.gradle.kts [hunk 2]
```diff
@@ -8,7 +8,7 @@ android {
 
   defaultConfig {
     applicationId = namespace
-    minSdk = 31
+    minSdk = libs.versions.minSdk.get().toInt()
```
Comment (saket) [inline sample/build.gradle.kts:11]: do you really need to run the sample app on all API levels? is having the library support 21 not sufficient?

File: gradle/libs.versions.toml (hunk 1 above)
Comment (KitsuneFolk) [reply gradle/libs.versions.toml]: Sure, I'll make a commit reverting this change tomorrow.

File: sample/build.gradle.kts (hunk 2 above)
Comment (KitsuneFolk) [reply sample/build.gradle.kts:11]: I tested the sample app on my real device, which is 29 API version, and I thought it would be better to downgrade the version for other people who might test it on lower API versions

Comment (saket) [reply sample/build.gradle.kts:11]: that's fair! let's keep this then.

File: gradle/libs.versions.toml (hunk 1 above)
Comment (KitsuneFolk) [reply gradle/libs.versions.toml]: The change is reversed.

**Review by saket — APPROVED** [review]
Thanks @MrRuslanYT!
